databases/week3/meal_reservation_database.sql

```diff
@@ -0,0 +1,143 @@
+
+/*Get all meals*/
+SELECT * FROM meal;
+
+/* add new meal*/
+INSERT INTO meal(`title`,`description`,`location`,`when`,`max_reservations`,`price`,`created_date`)
+ values('daal tadka', 'yellow indian daal','lyngby','2021-08-08 04:49:05','11','119.23','2020-08-08 20:40:23');
+
+--Meal queries
+ /*Get a meal with any id, fx 1*/
+ SELECT * FROM meal WHERE id = 2;
+
+ /*Update a meal with any id, fx 1. Update any attribute fx the title or multiple attributes*/
+UPDATE meal
+SET title = 'butter chicken', description = 'indian spicy buttter chicken masala' 
+WHERE id = 5;
+
+/*Delete a meal with any id, fx 1*/
+DELETE FROM meal WHERE id = 5;
+
+--reservation queries
+
+/*Add a new reservation*/
+INSERT INTO reservation(`number_of_guests`,`meal_id`,`contact_phonenumber`,`contact_name`,`contact_email`)
+VALUES('5','1','45897890','mark','mark@gmail.com'),('8','1','45897890','Kone','kone@gmail.com');
+/*Get a reservation with any id, fx 1*/
+SELECT * FROM reservation WHERE id=1;
+
+/*Update a review with any id, fx 1. Update any attribute fx the title or multiple attributes*/
+UPDATE reservation
+SET number_of_guests = '3', contact_phonenumber = '56789034' 
+WHERE id = 3;
+
+/*Delete a review with any id, fx 1*/
+
+DELETE FROM reservation WHERE id = 3;
+
+/*Get all reviews*/
+SELECT * FROM review;
+/*Add a new review*/
+insert into review(`title`,`description`,`meal_id`,`stars`)
+values('yummy','yummy spicy chicken curry','1','3');
+/*Get a review with any id, fx 1*/
+SELECT * FROM review WHERE id = 3;
+
+/*Update a review with any id, fx 1. Update any attribute fx the title or multiple attributes*/
+UPDATE review
+SET title = 'chicken curry', stars = '5' 
+WHERE id = 3;
+
+/*Delete a review with any id, fx 1*/
+DELETE FROM review WHERE id = 1;
+
+---Additional queries
+/*Get meals that has a price smaller than a specific price fx 90*/
+SELECT * FROM meal WHERE price > 120;
+
+/*Get meals that still has available reservations*/
+SELECT meal.title, meal.max_reservations, reservation.number_of_guests
+FROM meal
+JOIN reservation ON meal.id = reservation.meal_id
+WHERE meal.max_reservations > reservation.number_of_guests;
+SELECT * FROM reservation;
+/*Get meals that partially match a title. Rød grød med will match the meal with the title Rød grød med fløde*/
+SELECT title FROM meal WHERE title LIKE "%chicken curry%";
+
+/*Get meals that has been created between two dates*/
+SELECT * FROM meal WHERE created_date BETWEEN '2010-08-08' AND '2021-03-20';
+
+/*Get only specific number of meals fx return only 5 meals*/
+SELECT * FROM meal LIMIT 5;
+
+/*Get the meals that have good reviews*/
+SELECT meal.title, review.stars FROM meal
+JOIN review ON meal.id = review.meal_id 
+WHERE review.stars > 4;
+
+/*Get reservations for a specific meal sorted by created_date */
+SELECT meal.title, meal.created_date, reservation.contact_name from meal
+JOIN reservation ON meal.id = reservation.meal_id
+WHERE meal.title = 'chicken curry'
+ORDER BY created_date;
```

Comment:
In this query, you have to use the reservation table and join the meal table instead. And in where close use id so you can get the correct result.

Reply:
I fix it, I used meal.id = 2 now but ..we can also use meal.title = 'chicken curry' and output is same